programs/mango-v4/src/state/oracle.rs

```diff
@@ -317,9 +356,53 @@ pub fn oracle_state_unchecked(
                 oracle_type: OracleType::SwitchboardV1,
             }
         }
+        OracleType::OrcaCLMM => {
+            let usd_state = usdc_state_unchecked(usd_feed_opt)?;
+
+            let whirlpool = Whirlpool::try_deserialize(&mut &data[..]).unwrap();
```

Comment:
I'm curious about measuring the cost of this. It uses anchor's borsh deserialization to fill a bunch of fields we don't even read. The data looks to be in a fixed layout, we could probably zero-copy access it instead to save a bunch of deserialization overhead

Reply:
(since it's not aligned, we can't make a struct, but we could just manually take bytes N..M and intrepret them as the sqrt_price)

Reply:
Manually taking the bytes saves roughly 1000 CU